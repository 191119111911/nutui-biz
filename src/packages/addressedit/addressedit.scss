.nut-theme-dark {
    .nb-addressedit {
        background: $dark-background;
        color: $dark-color;
        &__item{ background: $dark-background;}
        .nut-switch.switch-open {
            background:$dark-color2;
        }
    }
    .nb-addressedit__bottom{
        background: $dark-background;
        .nut-button{
            color: $dark-color;
        }
        
    }
}
.nb-addressedit{
    background: $white;
    padding:5px 10px;
    border-radius: 4px;
    &__item{
        background: $white;
        padding: 0 10px;
        font-size: $font-size-2;
        color:$title-color2;
        &.setdefualt{
            padding: 10px 20px;
            display: flex;
            align-items: center;
            .label{
                margin-right: 20px;
            }
        }
    }
    &__bottom{
        position: fixed;
        left: 0;
        right: 0;
        bottom: 0;
        bottom: constant(safe-area-inset-bottom);
        bottom: env(safe-area-inset-bottom);
        width: 100%;
        padding: 12px 18px 24px;
        z-index: 999;
        background-color: #fff;
        box-sizing: border-box;
    }
    .nut-input{
        padding: var(--nutui-input-padding, 10px 10px);
    }
    .nut-input.nut-input-required:before{
        left: 0;
    }
    .nut-input-error-message{
        line-height: 1.2;
    }
}