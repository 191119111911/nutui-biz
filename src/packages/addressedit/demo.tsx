import React, { useState } from "react";
import {
  AddressEdit,
  AddressData,
  AddressInfo,
  AddressResult,
} from "./addressedit";
import { Radio, Cell, CellGroup, Input } from "@nutui/nutui-react";
import { useTranslate } from "../../sites/assets/locale";

const { RadioGroup } = Radio;

interface T {
  basic: string;
  selected: string;
  hasInfo: string;
  notInfo: string;
  custom1: string;
  custom2: string;
}

const AddressEditDemo = () => {
  const [translated] = useTranslate<T>({
    "zh-CN": {
      basic: "基本用法",
      selected: "已选地址",
      hasInfo: "修改地址",
      notInfo: "新增地址",
      custom1: "隐藏保存按钮",
      custom2: "自定义输入框",
    },
    "en-US": {
      basic: "Basic usage",
      selected: "Selected address",
      hasInfo: "Modify address",
      notInfo: "Add Address",
      custom1: "Hide Save Button",
      custom2: "Customize Input",
    },
  });

  const addressData: Partial<AddressResult> = {
    addressSelect: [],
    addressTitle: "选择所在地区",
    province: [
      { id: 1, name: "北京", title: "B" },
      { id: 2, name: "广西", title: "G" },
      { id: 3, name: "江西", title: "J" },
      { id: 4, name: "四川", title: "S" },
      { id: 5, name: "浙江", title: "Z" },
    ],
    city: [
      { id: 7, name: "朝阳区", title: "C" },
      { id: 8, name: "崇文区", title: "C" },
      { id: 9, name: "昌平区", title: "C" },
      { id: 6, name: "石景山区", title: "S" },
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 10, name: "北苑", title: "B" },
    ],
    country: [
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 9, name: "北苑", title: "B" },
      { id: 4, name: "常营乡", title: "C" },
    ],
    town: [],
    type: "custom",
  };
  const addressInfo: AddressInfo = {
    name: "",
    tel: "",
    region: "",
    regionIds: [],
    address: "",
    default: false,
  };
  const addressInfo2: AddressInfo = {
    name: "张三",
    tel: "13141234567",
    region: "北京朝阳区八里庄街道",
    regionIds: [1, 7, 3],
    address: "xxx小区3-2-302",
    default: true,
  };
  const addressSetData: Partial<AddressData> = {
    isRequired: ["name", "tel", "region", "address"],
    isDefualtAddress: true,
  };

  const addressSetData2: Partial<AddressData> = {
    nameText: "收件人",
    namePlaceholder: "请输入收件人姓名",
    isRequired: ["name", "tel"],
    isDefualtAddress: false,
    errorShowType: "errorMsg",
  };

  const addressData2: Partial<AddressResult> = {
    addressSelect: [1, 7, 3],
    addressTitle: "选择所在地区",
    province: [
      { id: 1, name: "北京", title: "B" },
      { id: 2, name: "广西", title: "G" },
      { id: 3, name: "江西", title: "J" },
      { id: 4, name: "四川", title: "S" },
      { id: 5, name: "浙江", title: "Z" },
    ],
    city: [
      { id: 7, name: "朝阳区", title: "C" },
      { id: 8, name: "崇文区", title: "C" },
      { id: 9, name: "昌平区", title: "C" },
      { id: 6, name: "石景山区", title: "S" },
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 10, name: "北苑", title: "B" },
    ],
    country: [
      { id: 3, name: "八里庄街道", title: "B" },
      { id: 9, name: "北苑", title: "B" },
      { id: 4, name: "常营乡", title: "C" },
    ],
    town: [],
    type: "elevator",
    height: "270px",
  };

  const onChange = (data: any) => {
    console.log("onChangeAddress", data);
  };

  const onClose = (data: any) => {
    console.log("onCloseAddress", data);
  };

  const [radioVal, setRadioVal] = useState("1");
  const handleChange = (v: any) => {
    setRadioVal(v);
  };



  return (
    <div className="demo" style={{ paddingBottom: "100px" }}>
      <>
        <CellGroup>
          <Cell>
            <RadioGroup
              value={radioVal}
              onChange={handleChange}
              style={{ flexFlow: "wrap" }}
            >
              <Radio value="1">{translated.notInfo}</Radio>
              <Radio value="2">{translated.hasInfo}</Radio>
              <Radio value="3">{translated.custom1}</Radio>
              <Radio value="4">{translated.custom2}</Radio>
            </RadioGroup>
          </Cell>
        </CellGroup>
        {/* <>{renderTpl(radioVal)}</> */}
        {radioVal === "1" && (
          <>
            <h2>{translated.basic}</h2>
            <AddressEdit
              address={addressData}
              data={addressSetData}
              addressInfo={addressInfo}
              onSave={(formData) => {
                console.log(formData);
              }}
            />
          </>
        )}
        {radioVal === "2" && (
          <>
            <h2>{translated.selected}</h2>
            <AddressEdit
              address={addressData2}
              data={addressSetData}
              addressInfo={addressInfo2}
              onSave={(formData) => {
                console.log(formData);
              }}
              onChangeAddress={onChange}
              onCloseAddress={onClose}
            />
          </>
        )}
        {radioVal === "3" && (
          <>
            <h2>{translated.custom1}</h2>
            <AddressEdit
              address={addressData2}
              data={addressSetData2}
              addressInfo={addressInfo2}
              showSave={false}
              onSwitch={(state, data) => {
                console.log("switch", state, data);
              }}
              onChange={(value, tag) => {
                console.log(tag, value);
              }}
              onCloseAddress={onClose}
            />
          </>
        )}
        {radioVal === "4" && (
          <>
            <h2>{translated.custom2}</h2>
            <AddressEdit
              address={addressData}
              data={addressSetData2}
              addressInfo={addressInfo}
              onChange={(value, tag) => {
                console.log(tag, value);
              }}
              onSave={(formData) => {
                console.log(formData);
              }}
              bottomInputTpl={
                <>
                  <div className="nb-addressedit__item">
                    <Input
                      label={"自定义内容1"}
                      className="nut-input-text"
                      defaultValue={""}
                      placeholder={"请输入"}
                      type="text"
                      clearable
                      onChange={(v, e) => {
                        console.log(v, e);
                      }}
                    />
                  </div>
                  <div className="nb-addressedit__item">
                    <Input
                      label={"自定义内容2"}
                      className="nut-input-text"
                      defaultValue={""}
                      placeholder={"请输入"}
                      type="text"
                      clearable
                      onChange={(v, e) => {
                        console.log(v, e);
                      }}
                    />
                  </div>
                </>
              }
            />
          </>
        )}
      </>
    </div>
  );
};

export default AddressEditDemo;
