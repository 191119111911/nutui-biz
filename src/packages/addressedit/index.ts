import { AddressEdit } from './addressedit'

export default AddressEdit
