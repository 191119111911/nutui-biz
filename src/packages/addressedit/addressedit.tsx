import React, { FunctionComponent, useState, useEffect } from "react";
import { useConfig } from "@/packages/configprovider";
import bem from "@/utils/bem";
import { IComponent } from "@/utils/typings";
import {
  Input,
  Button,
  Address,
  Switch,
  Cell,
  CellGroup,
} from "@nutui/nutui-react";

interface CalResult {
  type: string;
  data: any;
}
interface RegionData {
  name?: string;
  [key: string]: any;
}
interface AddressInfo {
  name?: string;
  tel?: string;
  region?: string;
  regionIds?: any;
  address?: string;
  default?: boolean;
}
interface AddressData {
  id?: string | number | any;
  nameText?: string;
  namePlaceholder?: string;
  nameErrorMsg?: string;
  telText?: string;
  telPlaceholder?: string;
  telErrorMsg?: string;
  telFormatter?: string;
  regionText?: string;
  regionPlaceholder?: string;
  regionErrorMsg?: string;
  addressText?: string;
  addressPlaceholder?: string;
  addressErrorMsg?: string;
  isDefualtAddress?: boolean;
  isRequired?: any;
  bottomText?: string;
}
interface AddressResult {
  addressSelect: any;
  addressStr: string;
  province: RegionData[];
  city: RegionData[];
  country: RegionData[];
  town: RegionData[];
  customAddressTitle: string;
  height?: string;
}

export interface AddressEditProps extends IComponent {
  addressInfo: AddressInfo;
  data: AddressData;
  address: any;
  addressType: string;
  onChange: (val: string, tag: string) => void;
  onChangeAddress: (data: any) => void;
  onCloseAddress: (data: any) => void;
  onSave: (data: any) => void;
}

const defaultProps = {
  addressInfo: {
    name: "",
    tel: "",
    region: "",
    regionIds: [],
    address: "",
    default: false,
  },
  data: {},
  addressType: "custom",
  address: {
    modelValue: false,
    addressSelect: [],
    province: [],
    city: [],
    country: [],
    town: [],
    customAddressTitle: "选择所在地区",
  },
} as unknown as AddressEditProps;

export const AddressEdit: FunctionComponent<
  Partial<AddressEditProps> &
    Omit<React.HTMLAttributes<HTMLDivElement>, "onChange">
> = (props) => {
  const { locale } = useConfig();
  const {
    addressInfo = null,
    showSave = false,
    data = null,
    address,
    addressType,
    onChange,
    onSave,
    onChangeAddress,
    onCloseAddress,
    ...rest
  } = {
    ...props,
  };
  //提交数据格式
  const [formData, setFormData] = useState<any>({
    name: "",
    tel: "",
    region: "",
    regionIds: [],
    address: "",
    default: false,
  });
  //必填项设置
  const [isRequired, setIsRequired] = useState([
    "name",
    "tel",
    "region",
    "address",
  ]);
  //地址组件相关数据
  const [addressData, setAddressData] = useState({
    addressSelect: [],
    province: [],
    city: [],
    country: [],
    town: [],
    addressTitle: "选择所在地区",
    height: "200px",
  });
  //地址编辑数据形式兜底文案配置
  const [editSeting, setEditSeting] = useState({
    nameText: "收货人",
    namePlaceholder: "请输入收货人",
    nameErrorMsg: "该项为必填项，请填写完后提交",
    telText: "手机号码",
    telPlaceholder: "请输入手机号码",
    telErrorMsg: "该项为必填项，请填写完后提交",
    regionText: "所在地区",
    regionPlaceholder: "请选择所在地区",
    regionErrorMsg: "该项为必填项，请填写完后提交",
    addressText: "详细地址",
    addressPlaceholder: "街道、楼牌号",
    addressErrorMsg: "该项为必填项，请填写完后提交",
    bottomText: "保存",
  });
  const [errorList, setErrorList] = useState<any>([]);
  const [showPopup, setShowPopup] = useState(false);
  const [type, setType] = useState("custom");

  useEffect(() => {
    //地址组件所需相关信息获取
    if (address) {
      setAddressData(address);
    }
    //编辑地址所需字段设置获取
    if (data) {
      setEditSeting({ ...data, ...editSeting });
      if (data?.isRequired) {
        setIsRequired(data.isRequired);
      }
    }
    //地址信息初始化获取
    if (addressInfo) {
      setFormData({ ...formData, ...addressInfo });
      //同步地址组件
      if (addressInfo.regionIds.length) {
        setAddressData({
          ...addressData,
          addressSelect: addressInfo.regionIds,
        });
      }
    }
    //重置error错误
    setErrorList([]);
  }, [address, data, addressInfo]);

  useEffect(() => {
    if (addressType) {
      setType(addressType);
    }
  }, [addressType]);

  const changeAddress = (cal: any, tag: string) => {
    setErrorList(errorList.filter((i: any) => i != "region"));
    if (cal.next === "town") {
      setShowPopup(false);
    }
    onChangeAddress && onChangeAddress(cal);
  };

  const closeAddress = (val: CalResult) => {
    if ((val.data as AddressResult).addressStr) {
      //地址id格式处理
      let ids = val.data.addressIdStr.split("_").map((item: any) => item * 1);
      //删除数组中为0的项
      ids.splice(ids.indexOf(0));
      //提交信息数据同步
      setFormData({ ...formData, region: val.data.addressStr, regionIds: ids });
      //地址组件数据同步
      setAddressData({ ...addressData, addressSelect: ids });
    }
    onCloseAddress && onCloseAddress(val);

    setShowPopup(false);
  };
  const inputOnchange = (val: any, tag: string) => {
    let data = { ...formData };
    if (val.length != 0) {
      setErrorList(errorList.filter((i: any) => i != tag));
    }

    Object.keys(formData).map((key: any) => {
      if (key === tag) {
        if (key === "tel") {
          const regTel =
            /^(13[0-9]|14[01456879]|15[0-35-9]|16[2567]|17[0-8]|18[0-9]|19[0-35-9])\d{8}$/;
          const regNumber = /[^-0-9]/g;
          let number =
            !regTel.test(val) && val.length > 11
              ? val.substring(0, 11)
              : val.replace(regNumber, "");
          data[tag] = number;
        } else {
          data[tag] = val;
        }
      }
    });
    setFormData({ ...formData, ...data });
    onChange && onChange(val, tag);
  };
  const validForm = () => {
    let form = { ...formData };
    let arr: any = [].concat(errorList);
    Object.keys(form).map((key) => {
      if (isRequired.includes(key) && form[key] === "") {
        switch (key) {
          case "name":
            if (!errorList.includes("name")) {
              arr.push("name");
            }
            break;
          case "tel":
            if (!errorList.includes("tel")) {
              arr.push("tel");
            }
            break;
          case "region":
            if (!errorList.includes("region")) {
              arr.push("region");
            }
            break;
          case "address":
            if (!errorList.includes("address")) {
              arr.push("address");
            }
            break;

          default:
            break;
        }
        setErrorList(arr);
        return false;
      }
    });

    return true;
  };
  const save = () => {
    if (validForm()) {
      onSave && onSave(formData);
    } else {
      console.log("校验不通过");
    }
  };
  const inputClear = (tag: string) => {
    let data = { ...formData };
    data[tag] = "";
    setFormData({ ...formData, ...data });
  };

  const b = bem("addressedit");
  return (
    <div className={`${b()}`} id={data?.id}>
      <div className={`${b("item")}`}>
        <Input
          className="nut-input-text"
          label={editSeting.nameText}
          defaultValue={formData.name}
          name="name"
          placeholder={editSeting?.namePlaceholder}
          type="text"
          clearable
          required={isRequired.includes("name") || false}
          onChange={(e) => inputOnchange(e, "name")}
          onClear={() => inputClear("name")}
          errorMessage={errorList.includes("name") && editSeting.nameErrorMsg}
        />
      </div>
      <div className={`${b("item")}`}>
        <Input
          className="nut-input-text"
          label={editSeting.telText}
          defaultValue={formData.tel}
          name="tel"
          placeholder={editSeting.telPlaceholder}
          type="tel"
          clearable
          required={isRequired.includes("tel") || false}
          onChange={(e) => inputOnchange(e, "tel")}
          onClear={() => inputClear("tel")}
          errorMessage={errorList.includes("tel") && editSeting.telErrorMsg}
        />
      </div>
      <div className={`${b("item")}`}>
        <Input
          className="nut-input-text"
          label={editSeting.regionText}
          defaultValue={formData.region}
          name="region"
          readonly
          placeholder={editSeting.regionPlaceholder}
          type="text"
          required={isRequired.includes("region") || false}
          onClick={() => setShowPopup(true)}
          errorMessage={
            errorList.includes("region") && editSeting.regionErrorMsg
          }
        />
        <Address
          modelValue={showPopup}
          type={type}
          modelSelect={addressData.addressSelect}
          province={addressData.province}
          city={addressData.city}
          country={addressData.country}
          town={addressData.town}
          height={addressData.height}
          customAddressTitle={addressData.addressTitle}
          onChange={(cal) => changeAddress(cal, "select")}
          onClose={closeAddress}
        />
      </div>
      <div className={`${b("item")}`}>
        <Input
          label={editSeting.addressText}
          className="nut-input-text"
          defaultValue={formData.address}
          placeholder={editSeting.addressPlaceholder}
          type="text"
          clearable
          required={isRequired.includes("address") || false}
          onChange={(e) => inputOnchange(e, "address")}
          onClear={() => inputClear("address")}
          errorMessage={
            errorList.includes("address") && editSeting.addressErrorMsg
          }
        />
      </div>
      <div className={`${b("item")} setdefualt`}>
        <span className="label">设置默认地址</span>
        <Switch
          checked={formData.default}
          onChange={(state) => {
            setFormData({ ...formData, default: state });
          }}
        />
      </div>
      <div className={`${b("bottom")}`}>
        <Button block type="danger" onClick={save}>
          {editSeting.bottomText}
        </Button>
      </div>
    </div>
  );
};

AddressEdit.defaultProps = defaultProps;
AddressEdit.displayName = "NutAddressEdit";
