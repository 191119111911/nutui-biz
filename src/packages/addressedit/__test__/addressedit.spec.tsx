import React from "react";
import { render, waitFor, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { AddressEdit } from "../addressedit";

const addressData: any = {
  addressSelect: [],
  addressTitle: "选择所在地区",
  province: [
    { id: 1, name: "北京", title: "B" },
    { id: 2, name: "广西", title: "G" },
    { id: 3, name: "江西", title: "J" },
    { id: 4, name: "四川", title: "S" },
    { id: 5, name: "浙江", title: "Z" },
  ],
  city: [
    { id: 7, name: "朝阳区", title: "C" },
    { id: 8, name: "崇文区", title: "C" },
    { id: 9, name: "昌平区", title: "C" },
    { id: 6, name: "石景山区", title: "S" },
    { id: 3, name: "八里庄街道", title: "B" },
    { id: 10, name: "北苑", title: "B" },
  ],
  country: [
    { id: 3, name: "八里庄街道", title: "B" },
    { id: 9, name: "北苑", title: "B" },
    { id: 4, name: "常营乡", title: "C" },
  ],
  town: [],
  type: "custom",
};
const addressInfo = {
  name: "",
  tel: "",
  region: "",
  regionIds: [],
  address: "",
  default: false,
};
const addressData2: any = {
  addressSelect: [1, 7, 3],
  addressTitle: "选择所在地区",
  province: [
    { id: 1, name: "北京", title: "B" },
    { id: 2, name: "广西", title: "G" },
    { id: 3, name: "江西", title: "J" },
    { id: 4, name: "四川", title: "S" },
    { id: 5, name: "浙江", title: "Z" },
  ],
  city: [
    { id: 7, name: "朝阳区", title: "C" },
    { id: 8, name: "崇文区", title: "C" },
    { id: 9, name: "昌平区", title: "C" },
    { id: 6, name: "石景山区", title: "S" },
    { id: 3, name: "八里庄街道", title: "B" },
    { id: 10, name: "北苑", title: "B" },
  ],
  country: [
    { id: 3, name: "八里庄街道", title: "B" },
    { id: 9, name: "北苑", title: "B" },
    { id: 4, name: "常营乡", title: "C" },
  ],
  town: [],
  type: "custom2",
  height: "270px",
};
const addressInfo2 = {
  name: "张三",
  tel: "13141234567",
  region: "北京朝阳区八里庄街道",
  regionIds: [1, 7, 3],
  address: "xxx小区3-2-302",
  default: true,
};
const addressSetData = {
  nameText: "收件人",
  namePlaceholder: "请输入收件人姓名",
  isRequired: ["name", "tel", "region", "address"],
  isDefualtAddress: true,
};

test("Add New Address", async () => {
  const { container } = render(
    <AddressEdit
      address={addressData}
      data={addressSetData}
      addressInfo={addressInfo}
    />
  );
  const requiredItem = container.querySelectorAll(".nut-input-required");
  await waitFor(() => {
    expect(requiredItem.length).toBe(4);
    expect(container.querySelectorAll(".label-string")[0].innerHTML).toBe(
      addressSetData.nameText
    );
    expect(container).toMatchSnapshot();
  });
});

test("edit address", async () => {
  const { container } = render(
    <AddressEdit
      address={addressData2}
      data={addressSetData}
      addressInfo={addressInfo2}
    />
  );

  await waitFor(() => {
    expect(container.querySelectorAll(".input-text")[0]).toHaveAttribute(
      "value",
      `${addressInfo2.name}`
    );
    expect(container.querySelectorAll(".input-text")[1]).toHaveAttribute(
      "value",
      `${addressInfo2.tel}`
    );
    expect(container.querySelectorAll(".input-text")[2]).toHaveAttribute(
      "value",
      `${addressInfo2.region}`
    );
    expect(container.querySelectorAll(".input-text")[3]).toHaveAttribute(
      "value",
      `${addressInfo2.address}`
    );
    expect(container.querySelector(".nut-switch")).toHaveClass("switch-open");
  });
});

test("event onChangeAddress test", async () => {
  const changeHandle = jest.fn();
  const closeHandle = jest.fn();
  const { container } = render(
    <AddressEdit
      address={addressData2}
      data={addressSetData}
      addressInfo={addressInfo2}
      onChangeAddress={changeHandle}
      onCloseAddress={closeHandle}
    />
  );

  await waitFor(() => {
    const inputEl = container.querySelectorAll(".nut-input-text")[2];
    inputEl && fireEvent.click(inputEl);

    const regionEl = container.querySelector(".nb-address ");

    expect(regionEl).toBeTruthy;

    const regionItem = regionEl?.querySelectorAll(
      ".nb-address__region-item"
    )[0];
    regionItem && fireEvent.click(regionItem);

    waitFor(() => {
      expect(changeHandle).toBeCalled();
      expect(closeHandle).toBeCalled();
    });
  });
});

test("event onSave test", async () => {
  const saveHandle = jest.fn();
  const { container } = render(
    <AddressEdit
      address={addressData2}
      data={addressSetData}
      addressInfo={addressInfo2}
      onSave={saveHandle}
    />
  );

  const bottomEl = container.querySelector(
    ".nb-addressedit__bottom .nut-button"
  ) as HTMLElement;
  fireEvent.click(bottomEl);
  expect(saveHandle).toBeCalled();
});
