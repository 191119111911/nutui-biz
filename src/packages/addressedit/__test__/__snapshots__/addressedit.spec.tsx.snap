// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add New Address 1`] = `
<div>
  <div
    class="nb-addressedit"
    id="undefined"
  >
    <div
      class="nb-addressedit__item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            收件人
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="name"
                placeholder="请输入收件人姓名"
                style="text-align: left;"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nb-addressedit__item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            手机号码
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="tel"
                placeholder="请输入手机号码"
                style="text-align: left;"
                type="tel"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nb-addressedit__item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            所在地区
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="region"
                placeholder="请选择所在地区"
                readonly=""
                style="text-align: left;"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="first-render hidden-render nut-overlay"
        style="z-index: 1000; animation-duration: 0.3s;"
      />
    </div>
    <div
      class="nb-addressedit__item"
    >
      <div
        class="nut-input nut-input-required nut-input-border  nut-input-text"
      >
        <div
          class="nut-input-label "
          style="width: 80px; text-align: left;"
        >
          <div
            class="label-string"
          >
            详细地址
          </div>
        </div>
        <div
          class="nut-input-value"
        >
          <div
            class="nut-input-main-con"
          >
            <div
              class="nut-input-inner"
            >
              <input
                class="input-text"
                maxlength="9999"
                name="address"
                placeholder="街道、楼牌号"
                style="text-align: left;"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nb-addressedit__item setdefualt"
    >
      <span
        class="label"
      >
        设置默认地址
      </span>
      <div
        class="nut-switch switch-close  nut-switch-base "
      >
        <div
          class="switch-button"
        >
          <div
            class="close-line"
          />
        </div>
      </div>
    </div>
    <div
      class="nb-addressedit__bottom"
    >
      <button
        class="nut-button nut-button--danger nut-button--normal nut-button--round nut-button--block "
      >
        <div
          class="nut-button__warp"
        >
          <div
            class=""
          >
            保存
          </div>
        </div>
      </button>
    </div>
  </div>
</div>
`;
